### PR TITLE
S19

### PR DESCRIPTION
**Pull request by devwistkey**

### CODE REVIEW

**Review by tralkan — CHANGES_REQUESTED** [review]

File: components/SSI/DIDxWallet/SocialRecovery/Recover/index.tsx [hunk 1]
```diff
@@ -142,11 +149,20 @@ function Component() {
             const tyron_: tyron.TyronZil.TransitionValue =
                 await tyron.Donation.default.tyron(donation)
 
-            const params = await tyron.TyronZil.default.DidSocialRecovery(
-                input,
-                txvalue,
-                tyron_
-            )
+            let params
+            if (parseFloat(resolvedInfo?.version?.slice(-5)!) >= 5.8) {
```
[the remark this answers is not in the document]
Comment (devwistkey) [reply components/SSI/DIDxWallet/SocialRecovery/Recover/index.tsx:205]: hi, yes

File: components/AddLiquidity/index.tsx [hunk 2]
```diff
@@ -264,11 +264,7 @@ function Component() {
         <>
             <div className={styles.container2}>
                 <div style={{ width: '30%' }}>
-                    <Selector
-                        option={option}
-                        onChange={handleOnChange}
-                        value={currency}
-                    />
+                    <Selector option={option} onChange={handleOnChange} />
```
[the remark this answers is not in the document]
Comment (devwistkey) [reply components/AddLiquidity/index.tsx:267]: It has default value from the selector itself @tralkan

**Review by tralkan — CHANGES_REQUESTED** [review]

File: components/SSI/DIDxWallet/SocialRecovery/Recover/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (devwistkey) [reply components/SSI/DIDxWallet/SocialRecovery/Recover/index.tsx:205]: I found DIDxWALLET_5.10.0 without 'v'. So I'll start from '_'

File: components/SSI/DIDxWallet/Wallet/DIDOperations/SocialRecovery/index.tsx [hunk 3]
```diff
@@ -115,14 +120,35 @@ function Component() {
...
 
     const handleSave = async () => {
+        setLoadingUserCheck(true)
         if (guardians.length === input_.length) {
-            for (let i = 0; i < guardians.length; i++) {
-                await resolveDid(guardians[i].toLowerCase())
+            var arr = guardians.map((v) => v.toLowerCase())
+            const duplicated = new Set(arr).size !== arr.length
+            if (duplicated) {
+                toast.error('Repeated username are not allowed', {
```
[the remark this answers is not in the document]
Comment (devwistkey) [reply components/SSI/DIDxWallet/Wallet/DIDOperations/SocialRecovery/index.tsx]: done

File: components/SSI/DIDxWallet/SocialRecovery/Recover/index.tsx [hunk 4]
```diff
@@ -100,15 +105,44 @@ function Component() {
         setHideDonation(true)
         setHideSubmit(true)
     }
+
+    const resolveDid = async (_username: string) => {
```
[the remark this answers is not in the document]
Comment (devwistkey) [reply components/SSI/DIDxWallet/SocialRecovery/Recover/index.tsx]: done

File: components/SSI/AddFunds/OriginatorAddress/index.tsx [hunk 5]
```diff
@@ -80,11 +80,14 @@ function Component() {
...
         if (input.includes('@')) {
-            username_ = input.split('@')[1].replace('.did', '').toLowerCase()
+            username_ = input
+                .split('@')[1]
+                .replace('.did', '')
+                .replace('.ssi', '')
+                .toLowerCase()
             domain_ = input.split('@')[0]
-        } else if (input.includes('.did')) {
+        } else if (input.includes('.did') || input.includes('.ssi')) {
```
[the remark this answers is not in the document]
Comment (devwistkey) [reply components/SSI/AddFunds/OriginatorAddress/index.tsx]: done

**Review by tralkan — APPROVED** [review]

File: components/SSI/DIDxWallet/SocialRecovery/Recover/index.tsx [hunk 6]
```diff
@@ -58,6 +63,23 @@ function Component() {
     const [input, setInput] = useState('') //the new address
     const [legend, setLegend] = useState('Save')
     const [button, setButton] = useState('button primary')
+    const [loadingInput, setLoadingInput] = useState(false)
+    const [mount, setMount] = useState(true)
+
+    const versionAbove58 = () => {
+        let res
+        var ver = resolvedInfo?.version?.split('_')[1]!
```
[the remark this answers is not in the document]
Comment (devwistkey) [reply components/SSI/DIDxWallet/SocialRecovery/Recover/index.tsx:71]: done

File: components/SSI/DIDxWallet/Wallet/DIDOperations/SocialRecovery/index.tsx [hunk 7]
```diff
@@ -50,24 +58,40 @@ function Component() {
...
     const { isController } = controller()
 
+    const domainNavigate =
+        resolvedInfo?.domain !== '' ? resolvedInfo?.domain + '@' : ''
+
     useEffect(() => {
         isController()
     })
 
+    const versionAbove58 = () => {
```
[the remark this answers is not in the document]
Comment (devwistkey) [reply components/SSI/DIDxWallet/Wallet/DIDOperations/SocialRecovery/index.tsx:74]: done

File: components/SSI/SBTxWallet/Wallet/index.tsx [hunk 8]
```diff
@@ -78,7 +78,10 @@ function Component({ type }) {
         let domain_ = ''
         if (input.includes('@')) {
             const [domain = '', username = ''] = input.split('@')
-            username_ = username.replace('.did', '').toLowerCase()
+            username_ = username
+                .replace('.did', '')
+                .replace('.ssi', '')
+                .toLowerCase()
```
[the remark this answers is not in the document]
Comment (devwistkey) [reply components/SSI/SBTxWallet/Wallet/index.tsx:84]: Hi, yes. I just tested it and it can resolve soul@tyron.ssi

**Pull request merged**